### ISSUE TITLE
Updating with a bad identifier throws 500 instead of 404

### ISSUE THREAD

**Issue by erickgonzalez** [dotCMS/core#15799]
Related to: https://github.com/dotCMS/core/issues/15433

When you update via curl and send an identifier that does not exist it returns a 500 and should return a 404.

```
curl -v -u admin@dotcms.com:admin -XPUT http://localhost:8080/api/vtl/employees -F "json={
   "identifier":"af047c97-1g91-4642-9cbb-2fbdef4da1cb",
   "contentType":"Employee",
   "languageId":1,
   "host1":"demo.dotcms.com",
   "firstName":"Daniel2",
   "lastName": "Silva2",
   "jobTitle": "Dev",
   "email": "daniel.silva@dotcms.com",
   "mobile": "8325553355", 
   "gender": "male",
   "binaryFields": [ "photo", "photo2" ]};
   type=application/json" -F "file=@/Users/danielsilva/Downloads/testImages/testImage1.jpg; type=application/jpg" -F "file=@/Users/danielsilva/Downloads/secondFile.jpg; type=application/jpg"
   
PATCH / UPDATE PARTIALLY

curl -v -u admin@dotcms.com:admin -XPATCH http://localhost:8080/api/vtl/employees/af047c97-1a91-4642-9gbb-2fbdef4da1cb -F "json={
   "lastName": "Silva Patched",
   "jobTitle": "Dev Patched",
   "binaryFields": [ "photo2" ]};
   type=application/json" -F "file=@/Users/danielsilva/Downloads/secondFile.jpg; type=application/jpg"   
```

**Comment by dsilvam** [comment]
The problem here is down at the API level - Workflow API, and one of the ways to reproduce it is via what's described above.

**Comment by dsilvam** [comment]
Stacktrace: https://gist.github.com/dsilvam/c6e9c8d191c982c17b58f3ea1e53afad

**Comment by jgambarios** [comment]
PR: https://github.com/dotCMS/core/pull/16305

**Comment by jgambarios** [comment]
PR: https://github.com/dotCMS/core/pull/16328

**Comment by bryanboza** [comment]
Fixed, new improvement reported on #16305
